toolchain/build-toolchain.sh

```diff
@@ -2,24 +2,39 @@
 # $1 - toolchain target (e.g. arm-phoenix)
 # $2 - toolchain install absolute path (i.e. no "." or ".." in the path)
 
-
 set -e
 
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
 log() {
     echo -e "\e[35;1m$*\e[0m"
 }
 
-if [ -z "$1" ]; then
-    echo "No toolchain target provided! Abort."
-    echo "officially supported targets: arm-phoenix i386-pc-phoenix riscv64-phoenix"
+# targets for libphoenix and phoenix-rtos-kernel for installing headers
+declare -A TOOLCHAN_TO_PHOENIX_TARGET=(
+    [arm-phoenix]="armv7a7-imx6ull"
+    [i386-pc-phoenix]="ia32-generic"
+    [riscv64-phoenix]="riscv64-spike"
+)
+
+if [ -z "$1" ] || [ -z "${TOOLCHAN_TO_PHOENIX_TARGET[$1]}" ]; then
+    echo "Missing or invalid target provided! Abort."
+    echo "officially supported targets:" "${!TOOLCHAN_TO_PHOENIX_TARGET[@]}"
     exit 1
 fi
 
+PHOENIX_TARGET="${TOOLCHAN_TO_PHOENIX_TARGET[$1]}"
+
 if [ -z "$2" ]; then
     echo "No toolchain install path provided! Abort."
     exit 1
 fi
 
+if [ ! -f "$SCRIPT_DIR/../../phoenix-rtos-kernel/Makefile" ] || [ ! -f "$SCRIPT_DIR/../../libphoenix/Makefile" ]; then
+    echo "phoenix-rtos-kernel or libphoenix missing! Please use full phoenix-rtos-project repo for toolchain building. Abort."
```

Comment:
Maybe create the variables `KERNEL_PATH` and `LIBPHOENIX_PATH`

Reply:
used only twice, no real benefit for now. Keeping as-is.